transmogrifier/sources/xml/ead.py

```diff
@@ -557,6 +493,96 @@ def get_publishers(cls, source_record: Tag) -> list[timdex.Publisher] | None:
             return [timdex.Publisher(name=publication_value)]
         return None
 
+    @classmethod
+    def get_related_items(cls, source_record: Tag) -> list[timdex.RelatedItem] | None:
+        related_items = []
+        collection_description = cls._get_collection_description(source_record)
+
+        for related_item_element in collection_description.find_all(
+            ["altformavail", "relatedmaterial", "separatedmaterial"], recursive=False
+        ):
+            if related_item_element.name == "relatedmaterial":
+                related_items.extend(cls._get_related_material(related_item_element))
+            else:  # noqa: PLR5501
+                if related_item := cls.create_string_from_mixed_value(
+                    related_item_element, separator=" ", skipped_elements=["head"]
+                ):
+                    related_items.append(
+                        timdex.RelatedItem(
+                            description=related_item,
+                            relationship=aspace_type_crosswalk.get(
+                                related_item_element.name, related_item_element.name
+                            ),
+                        )
+                    )
+        return related_items or None
+
+    @classmethod
+    def _get_related_material(
```

Comment:
Last comment on these related methods: I still don't know how, or if, there is a more straight-forward way to parse these from those wild semantic + display XML elements, but noting these feel like areas that could be cleaned up a bit.  

I think, for me, this refactor work is nice because it exposes some tricky corners like this, but are likely tricky for good reason because the logic is intricate.  I don't see any need to address now, and maybe not ever, but I'm appreciating having some insight into corners of these transformations I was not familiar with previously.

Reply:
Agreed this should have an issue for later review, it's rough (coming from the likely author 🙃 who knows better now)

Reply:
Taking another look though, could it skip the `related_materials = []` and  just return:

```
return [
                timdex.RelatedItem(description=related_item)
                for subelement in subelements
                if (
                    related_item := cls.create_string_from_mixed_value(
                        subelement, separator=" ", skipped_elements=["head"]
                    )
                )
            ]
```

Apologies if I'm going down the rabbit hole you already went down @ghukill !

Reply:
Tis a rabbit hole indeed!  When I saw the XML getting parsed, I immediately appreciated some of the logic here.  It's not easy to extract.

Your proposal above makes sense to me!  Does this mean that elements `["altformavail", "separatedmaterial"]` aren't needed?  and if so, is this true across all EAD records?

As an aside, this is precisely where I think many-record analysis could be extremely helpful.  If we could:
  * run against 100 EADs, save output
  * make change, run against same files, save output
  * have a convenient way to see the difference for this particular field

Lots of approaches come to mind, but the many-record comparison pattern itself is what I'm interested in.

Reply:
Agree on the utility of record analysis in a situation like this! I was just referring to updating the `_get_related_material` method, I believe `"altformavail` and `separatedmaterial` would still be caught in the updated `elif` block you recommended for `get_related_items`. But this is also a sign that this should be flagged in an issue for later consideration since it's complicated!

Reply:
@ehanson8 Made the change to remove `related_materials = []` from `_get_related_material()` (and directly return the result from list comprehension instead. 

@ghukill Many-record comparison sounds ideal! I think the `attributes_and_subfield_variations` fixture tries to address this, but it takes looking at the XML file to see the different conditions it tries to trigger. I definitely think a convenient way to see the outputs across many records and comparing different runs would be helpful for complex transforms!

Also, I tried to provide more context regarding the updates to `get_related_material` here https://github.com/MITLibraries/transmogrifier/pull/192#discussion_r1631457455! A long read but hoped to provide more clarification!